src/java.base/share/classes/java/lang/reflect/Field.java

```diff
@@ -174,7 +175,15 @@ Field copy() {
     @CallerSensitive
     public void setAccessible(boolean flag) {
         AccessibleObject.checkPermission();
-        if (flag) checkCanSetAccessible(Reflection.getCallerClass());
+        // Always check if the field is a final StableValue
+        if (StableValue.class.isAssignableFrom(type) && Modifier.isFinal(modifiers)) {
```

Comment:
I think we don't need these code change if we have `Field.isTrustedFinal()` return `true`; such fields' accessors don't support modifications.

Reply:
Actually, it should probably be done in `fieldDescriptor.cpp`’s [`bool fieldDescriptor​::is_trusted_final()`](https://github.com/openjdk/jdk/blob/34ee06f51122e8afb875cc3b75f513912272fd9b/src/hotspot/share/runtime/fieldDescriptor.cpp#L45-L48).

Reply:
Let me know what you think about the solution I just added